File: 1221. Split a String in Balanced Strings.py
class Solution:
    def balancedStringSplit(self, s: str) -> int:
        c, rc, lc = 0, 0, 0
        for i in s:
            if i == 'R':
                rc += 1
            else:
                lc += 1
            if rc == lc:
                c += 1
                rc, lc = 0, 0
        return(c)

File: 1221. Split a String in Balanced Strings.cpp
class Solution {
public:
    int balancedStringSplit(string s) {
        int c = 0, rc = 0, lc = 0;
        for(int i = 0; i < s.length(); i++){
            if (s[i] == 'R'){
                rc += 1;
            }
            else{
                lc += 1;
            }
            if (lc == rc){
                c += 1;
                lc = 0;
                rc = 0;
            }
        }
        return c;
    }
};
